master/internal/command/authz_basic_impl.go

```diff
@@ -42,6 +43,44 @@ func (a *NSCAuthZBasic) CanSetNSCsPriority(
 	return nil
 }
 
+// AccessibleScopes returns the set of scopes that the user should be limited to.
+func (a *NSCAuthZBasic) AccessibleScopes(
+	ctx context.Context, curUser model.User, requestedScope model.AccessScopeID,
+) (model.AccessScopeSet, error) {
+	var ids []int
+	returnScope := model.AccessScopeSet{}
+
+	if requestedScope == 0 {
```

Comment:
I think there is a case to be made here that we shouldn't need to go to the db in this case in the basic impl since we know we're not limiting the user. but the other options has their own downsides as well. this works for now imo

Reply:
yeah imo the best way to implement this function is have it return all the workspaces you have access to. That way there is no ambiguity between special values like nil or zero.